src/toxcore/dht/daemon_state.rs

```diff
@@ -143,17 +193,41 @@ mod tests {
         // test with incompleted serialized data
         let serialized_vec = DaemonState::serialize_old(&alice);
         let serialized_len = serialized_vec.len();
-        let deserial_result = DaemonState::deserialize_old(&alice, &serialized_vec[..serialized_len - 1]).wait();
-        assert_eq!(deserial_result, Err(DeserializeError::IncompleteData { needed: Needed::Size(55), data: serialized_vec[..serialized_len - 1].to_vec()}));
+        match DaemonState::deserialize_old(&alice, &serialized_vec[..serialized_len - 1]).wait() {
+            Ok(_) => assert!(false),
+            Err(error) => {
+                assert_eq!(*error.kind(), DeserializeErrorKind::IncompleteData { needed: Needed::Size(55), data: serialized_vec[..serialized_len - 1].to_vec() });
+            }
+        }
 
         // test with serialized data corrupted
         let serialized_vec = [42; 10];
-        let deserial_result = DaemonState::deserialize_old(&alice, &serialized_vec).wait();
-        assert_eq!(deserial_result, Err(DeserializeError::Deserialize { error: ErrorKind::Tag, data: serialized_vec.to_vec()}));
+        match DaemonState::deserialize_old(&alice, &serialized_vec).wait() {
+            Ok(_) => assert!(false),
```

Comment:
assert!(false) should be changed to panic, but even better use unpack! macro to happy coverage checker.